src/site/stages/build/plugins/inject-vale-lint.js

```diff
@@ -0,0 +1,58 @@
+/* eslint-disable no-continue, no-param-reassign */
+
+// Gets the results of vale lint and inject them into each page
+// This follows the pattern of the axe checker though that may not suit our needs
+// as vale has to run in some sort of container/cli (not client-side like axe-core)
+// QUESTION Should/can we wrap the vale binary with a node module, would that be useful?
+// QUESTION Should we run this before the MD is turned into HTML?
+function injectValeLint(buildOptions) {
+  return (files, metalsmith, done) => {
+    const isEnabled = !!buildOptions.lintPlainLanguage;
+
+    if (!isEnabled) {
+      done();
+      return;
+    }
+    console.log("injectValeLint(): enabled, inject vale lint");
+
+    // QUESTION should we call a child process here?
+    // the idea is to iterate over the content of Object.keys(parsedContent)
+    // for each object (HTML page?) we want to run "vale" then
+    // output the results of stdout into a "results" object which can be used to
+    // inject "warning" lines on the generated html page
+    var exec = require('child_process').exec;
+    var cmd = `vale ${filename}`;
+    const child = exec('vale', (err, stdout, stderr) => {
+      if (err) {
+        throw err;
+      }
+
+      // TODO capture stdout and inject it into the html?
+      console.log("exec testing childout", stdout);
+    });
```

Comment:
I'm actually not familiar with Vale, and being lazy, I'll just ask here - does it have to be executed from the server? Or can it be run from the browser? I'm also wondering how this affects the build time. I know it's only enabled if the flag is explicitly passed during local dev, but if it adds a lot of time to build the site, it might not have any usefulness for local.

Reply:
For context: The point of this was to only use it on the preview server, so it shouldn't affect our "real" build time. More like the page load time in preview.